app/models/user.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :sent_friend_requests, -> { where('status = 0') },
+           class_name: 'Friendship', foreign_key: 'from_user_id'
+  has_many :sent_friend_requests_users, through: :sent_friend_requests,
+                                        source: :to_user
+
+  has_many :received_friend_requests, -> { where('status = 0') },
+           class_name: 'Friendship', foreign_key: 'to_user_id'
+  has_many :received_friend_requests_users, through: :received_friend_requests,
+                                            source: :from_user
+
+  has_many :requested_friendships, -> { where('status = 1') },
+           class_name: 'Friendship', foreign_key: 'from_user_id'
+  has_many :requested_friends, through: :requested_friendships,
+                               source: :to_user
+
+  has_many :accepted_friendships, -> { where('status = 1') },
+            class_name: 'Friendship', foreign_key: 'to_user_id'
+  has_many :accepted_friends, through: :accepted_friendships,
+                              source: :from_user
+
+  has_many :author_posts, class_name: 'Post', foreign_key: 'post_id'
```

Comment:
Something's not right here, we are looking for a foreign_key in the `posts` table called `post_id` while what we mean is the user that created the post. Shouldn't this foreign_key be `user_id` ?

Reply:
It's just the posts that the user authored. Maybe posts is more clear